numpy/f2py/tests/test_callback.py

```diff
@@ -211,3 +211,27 @@ class TestF77CallbackPythonTLS(TestF77Callback):
     compiler-provided
     """
     options = ["-DF2PY_USE_PYTHON_TLS"]
+
+
+class TestF90Callback(util.F2PyTest):
+
+    suffix = '.f90'
+
+    code = """
```

Comment:
`textwrap.dedent` might make this look a bit better.

Reply:
`textwrap.dedent(code)` gives a visually identical result (it just removes the trailing spaces). So, I just indented the code for a better look (which might be subjective though).

Reply:
I think it would be good to add `textwrap.dedent` here anyway, so that people debugging this in future don't have to think about the leading spaces (which dedent removes)

Reply:
Using `textwrap.dedent` now. Note that this is possible only because the code is in free format, for fixed-format Fortran source, `textwrap.dedent` could not be used.

Reply:
Indeed - I made a PR a while ago that tried to use `dedent` for fixed-format, and realized it was a can of worms I should not have opened.